packages/scope-manager/src/referencer/Referencer.ts

```diff
@@ -669,13 +669,6 @@ class Referencer extends Visitor {
     // enum members can be referenced within the enum body
     this.scopeManager.nestTSEnumScope(node);
 
-    // define the enum name again inside the new enum scope
-    // references to the enum should not resolve directly to the enum
-    this.currentScope().defineIdentifier(
-      node.id,
-      new TSEnumNameDefinition(node.id, node),
-    );
```

Comment:
Previously we added an enum name variable within the enum scope.
This was a laziness thing in part to ensure that recursive references didn't prevent unused variable checks.
I have aligned enums with functions and namespaces and removed this check and instead we rely on an explicit "is recursive" check (see changes to `collectUnusedVariables.ts`)